models/repo_mirror.go

```diff
@@ -264,6 +264,21 @@ func SyncMirrors() {
 			log.Error(4, "UpdateMirror [%s]: %v", repoID, err)
 			continue
 		}
+
+		// Get latest commit date and compare to current repository updated time,
+		// update if latest commit date is newer.
+		commitDate, err := git.GetLatestCommitTime(m.Repo.RepoPath())
+		if err != nil {
+			log.Error(2, "GetLatestCommitDate [%s]: %v", m.RepoID, err)
+			continue
+		} else if commitDate.Before(*m.Repo.UpdatedUnix.AsTimePtr()) {
```

Comment:
1) This will not fix already invalid values
2) If mirrored repo last commit is deleted with force push, this will not update to new latest commits time

Can't this be checked so to skip only when updated_unix equals commit time?

Reply:
@lafriks You are right. Removed the following check.

```go
if commitDate.Before(*m.Repo.UpdatedUnix.AsTimePtr())
```